src/PointwiseFunctions/AnalyticData/GrMhd/MagnetizedFmDisk.cpp

```diff
@@ -0,0 +1,320 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticData/GrMhd/MagnetizedFmDisk.hpp"
+
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <ostream>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.tpp"  // IWYU pragma: keep
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"  // IWYU pragma: keep
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/PolytropicFluid.hpp"  // IWYU pragma: keep
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/Requires.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+/// \cond
+namespace grmhd {
+namespace AnalyticData {
+
+MagnetizedFmDisk::MagnetizedFmDisk(
+    const double bh_mass, const double bh_dimless_spin,
+    const double inner_edge_radius, const double max_pressure_radius,
+    const double polytropic_constant, const double polytropic_exponent,
+    const double threshold_density, const double inverse_plasma_beta,
+    const size_t normalization_grid_res) noexcept
+    : bh_mass_(bh_mass),
+      bh_spin_a_(bh_mass * bh_dimless_spin),
+      inner_edge_radius_(bh_mass * inner_edge_radius),
+      max_pressure_radius_(bh_mass * max_pressure_radius),
+      polytropic_constant_(polytropic_constant),
+      polytropic_exponent_(polytropic_exponent),
+      threshold_density_(threshold_density),
+      inverse_plasma_beta_(inverse_plasma_beta),
+      normalization_grid_res_(normalization_grid_res),
+      fm_disk_{bh_mass,
+               bh_dimless_spin,
+               inner_edge_radius,
+               max_pressure_radius,
+               polytropic_constant,
+               polytropic_exponent},
+      equation_of_state_(fm_disk_.equation_of_state()),
+      background_spacetime_{
+          bh_mass_, {{0.0, 0.0, bh_dimless_spin}}, {{0.0, 0.0, 0.0}}},
+      kerr_schild_coords_{bh_mass, bh_dimless_spin} {
+  ASSERT(threshold_density_ > 0.0 and threshold_density_ < 1.0,
```

Comment:
This ASSERT is for the case where this is not constructed via the option parser (and to catch bugs, of course) because the parser already enforces these limits.